### PR TITLE
Add empower-functions integration to litellm

### PR DESCRIPTION
**Pull request by liuyl**
empower-functions is a function-calling model that is 3X faster and 10X cheaper than GPT-4, read more about the model: https://docs.empower.dev/inference/tool-use/introduction

This PR adds the empower-functions model to litellm.

Tested: normal conversation works with this PR

TODO:
* Test function calls. It is not possible right now, because I need to check in the `model_prices_and_context_window.json` file to let litellm recognize `empower-functions` model as a function-calling model.
* documentation for the model, will be added in the next PR, after I tested the function calling feature.

### CODE REVIEW

**Comment by ghost** [comment]
Hi @liuyl if it's openai-compatible, not sure we need a separate file for this. See groq implementation - https://github.com/BerriAI/litellm/pull/2168

**Comment by liuyl** [comment]
Hi @krrishdholakia, thanks for the context. I have deleted the empower.py file, please take another look.

File: litellm/__init__.py [hunk 1]
```diff
@@ -366,6 +371,11 @@ def identify(event_details):
     "meta-llama/Llama-2-70b-chat",
 ]  # these have been tested on extensively. But by default all text2text-generation and text-generation models are supported by liteLLM. - https://docs.litellm.ai/docs/providers
 
+empower_models = [
```
Comment (ghost) [inline litellm/__init__.py:371]: @liuyl this would override the models pulled from the model_prices_and_context_window.json

**Comment by ghost** [comment]
Hey @liuyl this looks fine - are you sure you don't want to add empower to the docs? https://docs.litellm.ai/docs/providers/groq

**Pull request closed without merging**